com.xargon.retrodeck.yml

```diff
@@ -0,0 +1,644 @@
+  - name: portaudio
+    buildsystem: cmake-ninja
+    config-opts:
+      - '-DCMAKE_BUILD_TYPE=Release'
+    cleanup:
+      - /include
+      - /share/doc
+      - /lib/cmake
+      - /lib/pkgconfig
+      - /lib/*.a
+    sources:
+      - type: git
+        url: https://github.com/PortAudio/portaudio.git
+        tag: v19.7.0
+        commit: 147dd722548358763a8b649b3e4b41dfffbcfbb6
+        x-checker-data:
+          type: git
+          tag-pattern: ^v([\d.]+)$
+  - name: soundtouch
+    buildsystem: cmake-ninja
+    sources:
+      - type: git
+        url: https://codeberg.org/soundtouch/soundtouch.git
+        tag: 2.3.1
+        commit: e1f315f5358d9db5cee35a7a2886425489fcefe8
+        x-checker-data:
+          type: git
+          tag-pattern: ^([\d.]+)$
+    cleanup:
+      - /bin
+      - /include
+      - /lib/cmake
+      - /lib/pkgconfig
+      - /share/doc
+      - /lib/*.a
+  - name: png++
+    disabled: true
+    buildsystem: simple
+    build-commands:
+      - make
+      - make prefix=/app install
+    sources:
+      - type: archive
+        url: >-
+          https://download.savannah.nongnu.org/releases/pngpp/png++-0.2.10.tar.gz
+        sha256: 998af216ab16ebb88543fbaa2dbb9175855e944775b66f2996fc945c8444eee1
+  - name: libpcap
+    buildsystem: cmake-ninja
+    sources:
+      - type: git
+        url: https://github.com/the-tcpdump-group/libpcap.git
+        tag: libpcap-1.10.1
+        commit: c7642e2cc0c5bd65754685b160d25dc23c76c6bd
+        x-checker-data:
+          type: git
+          tag-pattern: ^libpcap-([\d.]+)$
+    cleanup:
+      - /bin
+      - /include
+      - /share/man
+      - /lib/pkgconfig
+      - /lib/*.a
+  - name: pcsx2
+    buildsystem: cmake-ninja
+    builddir: true
+    config-opts:
+      - '-DCMAKE_BUILD_TYPE=Release'
+      - '-DPACKAGE_MODE=TRUE'
+      - '-DXDG_STD=TRUE'
+      - '-DDISABLE_BUILD_DATE=TRUE'
+      - '-DDISABLE_PCSX2_WRAPPER=TRUE'
+      - '-DDISABLE_ADVANCE_SIMD=TRUE'
+      - '-DSDL2_API=TRUE'
+    cleanup:
+      - /share/pixmaps
+      - /share/man
+      - /share/doc
+    post-install:
+      - >-
+        desktop-file-edit --set-key=Exec --set-value=pcsx2
+        /app/share/applications/PCSX2.desktop
+      - >-
+        install -Dm644 ../AppIcon128.png
+        /app/share/icons/hicolor/128x128/apps/PCSX2.png
+      - >-
+        install -Dm644 ../net.pcsx2.PCSX2.metainfo.xml
+        /app/share/metainfo/net.pcsx2.PCSX2.metainfo.xml
+    sources:
+      - type: git
+        url: https://github.com/PCSX2/pcsx2.git
+        tag: v1.7.2554
+        commit: 0d1675fbaa0d40874df98eed6270efc55620271b
+      - type: file
+        path: rd-submodules/pcsx2/net.pcsx2.PCSX2.metainfo.xml
+      - type: file
+        path: rd-submodules/pcsx2/AppIcon128.png
+
+  # pcsx2 - END
+
+
+  # Ryujinx
+  # Xemu
+  # Rpcs3
+  # vita3k
+  # dolphin
+  # arduboy
+  # pocket station
+  # cemu - win only: wine
+  # Anbox?
+
+
+  # External manifests end
+  
+
+  - name: retrodeck
```

Comment:
all of this should come from an "upstream" project.

Flathub is not meant to be hosting application. Just manifest to build then as flatpak.

Reply:
For many reasons I choose to don't upstream those files as it will break all the git actions, building and cooker channel.
Unfortunately this project is pretty much complicated comparing it to the average, so I ask you to please accept this method.

Maybe in the future I may find a clean way to upstream those, but now they are needed as they are.

Reply:
I don't see a source tarball or git. So I assume this is all. Hence my comment still stands.

Reply:
To be clear, as I mentioned it on Matrix already – Flathub organization is meant to hold only files needed for building a flatpak, and anything else should be hosted elsewhere.

Reply:
My build system is directly referring to those files, if I pack them then the stable and developer build will clash.
In order to edit this I should rething all the logics, but I don't have a better solution that, so I am sorry if I am putting more files than usual.